src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -343,12 +342,18 @@ private void AddNameAndTypeArgumentsOrParameters(INamedTypeSymbol symbol)
                 symbolName = symbol.Name;
             }
 
+            var isAnoymousDelegate = symbol.TypeKind == TypeKind.Delegate && !symbol.CanBeReferencedByName;
             if (format.MiscellaneousOptions.IncludesOption(SymbolDisplayMiscellaneousOptions.UseErrorTypeSymbolName) &&
                 partKind == SymbolDisplayPartKind.ErrorTypeName &&
                 string.IsNullOrEmpty(symbolName))
             {
                 builder.Add(CreatePart(partKind, symbol, "?"));
             }
+            else if (isAnoymousDelegate)
+            {
+                var name = $"<anonymous delegate>";
```

Comment:
Does this need to be localized?

Reply:
no.  we do the same here: https://github.com/dotnet/roslyn/blob/5bce330dc6294f059e6cf69a2ee77b02e141c804/src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs#L443-L447